### PR TITLE
Add kube-prometheus-stack chart (replaces prometheus-operator)

### PR DESCRIPTION
**Pull request by branden**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

This renames (copies) the prometheus-operator chart to kube-prometheus-stack, upgrades it to the latest version 15.4.4, and deprecates prometheus-operator.

This also includes some fixes to the chart upgrade scripts, see commits at the base of the branch.

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-75500

**Special notes for your reviewer**:

I tested that the prometheus addon successfully upgrades from prometheus-operator to kube-prometheus-stack.

The kube-prometheus-stack upgrade updates the apiversion for ingresses on k8s 1.19+ to `networking.k8s.io/v1`, which introduces a required `pathType` parameter to its spec. The chart does not set a default `pathType` for its ingresses, so it must be provided in chart values when deploying to k8s 1.19+. `pathType` support in ingresses was introduced with apiversion `networking.k8s.io/v1beta1` in k8s 1.18. This chart is used in an addon that is distributed with Konvoy releases that support k8s 1.16, so this may require separate prometheus addons for k8s < 1.18 and k8s >= 1.18.

**Does this PR introduce a user-facing change?**:

```release-note
[staging/prometheus-operator] Deprecated in favor of staging/kube-prometheus-stack.
[staging/kube-prometheus-stack] Chart added at version 15.4.4. This replaces prometheus-operator.
```

**Checklist**

* [x] *If a chart is changed, the chart version is correctly incremented.*
* [x] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

**Comment by branden** [comment]
It looks like CI is failing because the chart is new?

```
2021-05-06T23:36:37.154383638Z Error: could not find ct_previous_revision385282594/staging/kube-prometheus-stack: stat ct_previous_revision385282594/staging/kube-prometheus-stack: no such file or directory
2021-05-06T23:36:37.166182942Z Error building dependencies for previous revision of chart 'kube-prometheus-stack => (version: "15.4.4", path: "staging/kube-prometheus-stack")'
2021-05-06T23:36:37.166227497Z : Error waiting for process: exit status 1
```

**Review by gracedo — APPROVED** [review]
🥳 I'll prob take a look at the changelog at some point and see if there are any updates we can take advantage of

File: staging/kube-prometheus-stack/UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+## Upgrading
+
+To upgrade the operator, simply run:
+```sh
+./upgrade_operator.sh
+```
+
+The upgrade script:
+- clones the helm/charts repo `master` branch
+- copies all the files under `stable/prometheus-operator`
```
Comment (gracedo) [inline staging/kube-prometheus-stack/UPGRADING.md:14]:
```suggestion
- copies all the files under `stable/kube-prometheus-stack`
```

File: staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    use exiting storage config if it already exists
+    
+    If either prometheus or alertmanager resources exist and have a storage
+    attribute defined in their spec, reuse that instead of what's defined
+    in the values.
+    
+    This prevents the accidental redefinition of the volumeClaimTemplate and
+    the abaondomnet of old data.
+
+diff --git a/staging/prometheus-operator/templates/NOTES.txt b/staging/prometheus-operator/templates/NOTES.txt
```
Comment (gracedo) [inline staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch:14]:
```suggestion
diff --git a/staging/kube-prometheus-stack/templates/NOTES.txt b/staging/kube-prometheus-stack/templates/NOTES.txt
```

File: staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+index 371f3ae39..826ec6eb9 100644
+--- a/staging/prometheus-operator/templates/NOTES.txt
++++ b/staging/prometheus-operator/templates/NOTES.txt
```
Comment (gracedo) [inline staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch:17]:
```suggestion
--- a/staging/kube-prometheus-stack/templates/NOTES.txt
+++ b/staging/kube-prometheus-stack/templates/NOTES.txt
```

File: staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
++If this is not the desired behavior, delete the {{ $name }} alertmanager resource from the {{ $namespace }} namespace and upgrade this helm release.
++{{- end }}
++
+ {{ $.Chart.Name }} has been installed. Check its status by running:
+   kubectl --namespace {{ template "kube-prometheus-stack.namespace" . }} get pods -l "release={{ $.Release.Name }}"
+ 
+diff --git a/staging/prometheus-operator/templates/alertmanager/alertmanager.yaml b/staging/prometheus-operator/templates/alertmanager/alertmanager.yaml
+index 78d0f7c5b..9f0602e8d 100644
+--- a/staging/prometheus-operator/templates/alertmanager/alertmanager.yaml
++++ b/staging/prometheus-operator/templates/alertmanager/alertmanager.yaml
```
Comment (gracedo) [inline staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch:43]:
```suggestion
diff --git a/staging/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml b/staging/prometheus-operator/templates/alertmanager/alertmanager.yaml
index 78d0f7c5b..9f0602e8d 100644
--- a/staging/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml
+++ b/staging/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml
```

File: staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
++{{- else }}
++  storage: {{ toYaml .Values.alertmanager.alertmanagerSpec.storage | nindent 4 }}
++{{- end }}
+ {{- end }}
+ {{- if .Values.alertmanager.alertmanagerSpec.podMetadata }}
+   podMetadata:
+diff --git a/staging/prometheus-operator/templates/prometheus/prometheus.yaml b/staging/prometheus-operator/templates/prometheus/prometheus.yaml
+index eb316d64d..ddd7a8d06 100644
+--- a/staging/prometheus-operator/templates/prometheus/prometheus.yaml
++++ b/staging/prometheus-operator/templates/prometheus/prometheus.yaml
```
Comment (gracedo) [inline staging/kube-prometheus-stack/patch/mesosphere/patch/11_use_existing_storage_definitions.patch:68]:
```suggestion
diff --git a/staging/kube-prometheus-stack/templates/prometheus/prometheus.yaml b/staging/kube-prometheus-stack/templates/prometheus/prometheus.yaml
index eb316d64d..ddd7a8d06 100644
--- a/staging/kube-prometheus-stack/templates/prometheus/prometheus.yaml
+++ b/staging/kube-prometheus-stack/templates/prometheus/prometheus.yaml
```

File: staging/kube-prometheus-stack/patch/patch_9_mesosphere_helmignore.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+set -xeuo pipefail
+
+SRCFILE="${BASEDIR}"/.helmignore
+
+sed -i '/# Mesosphere-specific files to ignore/,$d' ${SRCFILE}
+
+cat << EOF >> ${SRCFILE}
+# Mesosphere-specific files to ignore
+upgrade_operator.sh
+prometheus-operator-*.tgz
```
Comment (gracedo) [inline staging/kube-prometheus-stack/patch/patch_9_mesosphere_helmignore.sh:16]:
```suggestion
```

**Comment by alejandroEsc** [comment]
I am seeing these errors: ```Error: no repository definition for https://prometheus-community.github.io/helm-charts, https://prometheus-community.github.io/helm-charts, https://grafana.github.io/helm-charts. Please add the missing repos via 'helm repo add'``` if they are required you need to add them to ct configs in the test directory iirc

**Comment by branden** [comment]
Looks like that error is happening because a couple `ct.yaml`s omit `prometheus-operator` by name, so I just need to update those references to use the new chart name. Commit on the way.

**Comment by alejandroEsc** [comment]
Awesome work. Thank you for testing it out and fixing the CI issue.

**Review by alejandroEsc — APPROVED** [review]
lgtm, again thanks for testing!

**Pull request merged**